### PR TITLE
improvement: new limit checker layout, blotter horizontal scroll

### PR DESCRIPTION
**Pull request by SHolleworth**
Updated limit checker

Screenshot 2023-06-22 at 18 13 46
  
Horizontal blotter scrolling. (also applied to limit checker result table)

Screenshot 2023-06-22 at 18 13 30

### CODE REVIEW

**Comment by SHolleworth** [comment]
An issue with the scrolling blotter, the vertical scroll bar is out of view when horizontal content is hidden to the right. This obviously isn't ideal but i have yet to figure out a way to have vertical and horizontal scroll on the same container with react-window

File: src/client/src/App/Trades/TradesGrid/styled.ts [hunk 1]
```diff
@@ -8,6 +8,8 @@ export const Table = styled.div`
   border-bottom: 15px solid ${({ theme }) => theme.core.darkBackground};
   height: calc(100% - 4.75rem);
   width: 100%;
+  overflow-x: scroll;
```
Comment (snony14) [inline src/client/src/App/Trades/TradesGrid/styled.ts:11]: Do we have to display the scroll constantly, or maybe it is better to be auto-displayed?

File: src/client/src/OpenFin/apps/LimitChecker/LimitInput.tsx [hunk 2]
```diff
@@ -1,16 +1,47 @@
...
 
 import { setLimit, useLimit } from "./state"
 
 const Container = styled.div`
-  display: flex;
-  justify-content: center;
-  align-items: center;
-  background-color: ${({ theme }) => theme.core.darkBackground};
+  background-color: ${({ theme }) => theme.core.lightBackground};
   padding: 10px 10px 10px 10px;
```
Comment (snony14) [inline src/client/src/OpenFin/apps/LimitChecker/LimitInput.tsx:9]: `padding: 10px` is the same as the expression.

File: src/client/src/OpenFin/apps/LimitChecker/LimitInputList.tsx [hunk 3]
```diff
@@ -6,21 +6,19 @@ import { LimitInput } from "./LimitInput"
...
 
-const Grid = styled.div`
-  display: grid;
-  grid-template-columns: 1fr 1fr 1fr;
-  grid-gap: 3px;
-  padding: 10px 10px 10px 10px;
-  border-radius: 3px;
+const Scroll = styled.div`
+  padding-left: 10px;
+  overflow: scroll;
```
Comment (snony14) [inline src/client/src/OpenFin/apps/LimitChecker/LimitInputList.tsx:21]: `overflow: auto`?

File: src/client/src/OpenFin/apps/LimitChecker/LimitInput.tsx [hunk 4]
```diff
@@ -1,16 +1,47 @@
...
+  margin-bottom: 10px;
+
+  &:hover {
+    background-color: ${({ theme }) => theme.core.backgroundHoverColor};
+  }
+`
+
+const Symbol = styled.div`
+  font-size: 0.8125rem;
```
[the remark this answers is not in the document]
Comment (algreasley) [reply src/client/src/OpenFin/apps/LimitChecker/LimitInput.tsx:18]: (same category of comment as @snony14 above, re padding expression, needs an overhaul, let's add/augment a ticket)

File: src/client/src/App/Trades/TradesGrid/styled.ts (hunk 1 above)
Comment (SHolleworth) [reply src/client/src/App/Trades/TradesGrid/styled.ts:11]: For some reason, the bar _is_ being auto hidden even though the overflow is not set to auto. I will change the CSS in any case

File: src/client/src/OpenFin/apps/LimitChecker/LimitInput.tsx (hunk 2 above)
Comment (SHolleworth) [reply src/client/src/OpenFin/apps/LimitChecker/LimitInput.tsx:9]: Changed

File: src/client/src/App/Trades/TradesGrid/styled.ts (hunk 1 above)
Comment (SHolleworth) [reply src/client/src/App/Trades/TradesGrid/styled.ts:11]: On investigation it seems like the scroll bar is still there but the handle dissapears when the content isn't clipped

File: src/client/src/OpenFin/apps/LimitChecker/LimitInputList.tsx (hunk 3 above)
Comment (SHolleworth) [reply src/client/src/OpenFin/apps/LimitChecker/LimitInputList.tsx:21]: changed

File: src/client/src/OpenFin/apps/LimitChecker/LimitInput.tsx [hunk 5]
```diff
@@ -1,16 +1,47 @@
...
+  outline: none;
+  border: none;
+  font-size: 0.75rem;
+
+  padding: 2px 0;
+  color: ${({ theme }) => theme.core.textColor};
+  border-bottom: 1.5px solid ${({ theme }) => theme.primary[5]};
+  caret-color: ${({ theme }) => theme.primary.base};
+  &:focus {
+    outline: none !important;
```
[the remark this answers is not in the document]
Comment (SHolleworth) [reply src/client/src/OpenFin/apps/LimitChecker/LimitInput.tsx:42]: seems to work fine without the !important, it was copy and pasted over from the tile inputs

**Review by algreasley — APPROVED** [review]

**Pull request merged**